src/psmove.c

```diff
@@ -811,12 +841,19 @@ psmove_connect_by_id(int id)
 #else
     if (id < available) {
         cur_dev = devs_sorted[id];
-        move = psmove_connect_internal(cur_dev->serial_number, cur_dev->path, id);
+        move = psmove_connect_internal(cur_dev->serial_number, cur_dev->path, id, cur_dev->product_id);
     }
 #endif
 
     free(devs_sorted);
-    hid_free_enumeration(devs);
+
+    // free HID device enumerations
+    for (i = 0; i < NUM_PSMOVE_PIDS; i++) {
+        devs = hid_enumerate(PSMOVE_VID, PSMOVE_PIDS[i]);
```

Comment:
This shouldn't do a `hid_enumerate()` again, but just a `hid_free_enumeration(move_hid_devices[i]);`, right?